src/Controller/Controller.php

```diff
@@ -227,8 +227,9 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * @param \Cake\Network\Response|null $response Response object for this controller.
      * @param string|null $name Override the name useful in testing when using mocks.
      * @param \Cake\Event\EventManager|null $eventManager The event manager. Defaults to a new instance.
+     * @param \Cake\Controller\ComponentRegistry|null $components The component registry. Defaults to a new instance.
      */
-    public function __construct(Request $request = null, Response $response = null, $name = null, $eventManager = null)
+    public function __construct(Request $request = null, Response $response = null, $name = null, $eventManager = null, $components = null)
     {
```

Comment:
What benefits it will give to us? As for me - flat dependencies is the best solution. Moreover I afraid it's not a part of this task.